### ISSUE TITLE
Track names used within a single test file

### ISSUE THREAD

**Issue by schloerke** [rstudio/shinytest2#137]
Since we are wrapping `testthat::test_dir()`, we can wrap `testthat::test_dir(reporter=)` which has access to [`reporter$start_file(file)`](https://github.com/r-lib/testthat/blob/22a68dd85b65c95173424760a78955b6c55f3848/R/reporter-progress.R#L80-L86) / [`$end_file(file)`](https://github.com/r-lib/testthat/blob/22a68dd85b65c95173424760a78955b6c55f3848/R/reporter-progress.R#L429-L442).

Idea:
* Wrap this reporter with a `{shinytest2}` shim to capture the file value.
* When saving a snapshot file, check for the known names that have been seen if the reporter flag is set.
* Ignore if the flag is not set

Possible problems:
* Executing tests in parallel

**Comment by schloerke** [comment]
Marking this issue as `wontfix` as there is no good fix but to fix it upstream in `{testthat}`
